users/dierkes/preprocessing/segment_cutting.py

```diff
@@ -15,42 +18,91 @@ class CutAndStitchSpeechSegmentsFromCorpusJob(Job):
 
 
     """
-    def __init__(self, bliss_corpus_file, target_length=20, n_workers=4, file_extension='wav'):
+
+    def __init__(
+        self,
+        bliss_corpus_file,
+        min_length=10,
+        target_length=30,
+        n_workers=4,
+        file_extension="wav",
+    ):
         """
 
         :param tk.Path bliss_corpus:
+        :param int min_length: minimal length of the stitched extracted speech in seconds. Note that
+            this constraint might be violated in some cases, e.g. when a recording contains less
+            speech than min_length or because the greedy stitching strategy might produce leftover
+            files for every recording shorter than min_length
         :param int target_length: target length of the stitched extracted speech in seconds
         :param int n_workers: number of threads for work on files in parallel
         :param str file_extension: currently supported are wav, flac and mp3
         """
         self.bliss_corpus_file = bliss_corpus_file
+        self.min_length = min_length
         self.target_length = target_length
         self.n_workers = n_workers
         self.file_extension = file_extension
-        assert file_extension in ['wav', 'flac', 'mp3']
+        assert file_extension in ["wav", "flac", "mp3"]
 
         self.out_audio_path = self.output_path("audio/", directory=True)
-        self.rqmt = {'time': 8, 'mem': 16, 'cpu': self.n_workers}
+        self.out_length_hist = self.output_path("file_length_hist.png")
+        self.cut_and_stitch_segments_rqmt = {
+            "time": 8,
+            "mem": 16,
+            "cpu": self.n_workers,
+        }
+        self.length_dist_rqmt = {"time": 2, "mem": 1, "cpu": self.n_workers}
```

Comment:
```suggestion
        self.length_dist_rqmt = {"time": 2, "mem": 1, "cpu": 1}
```
do you really need more than 1 worker for plotting? This will lead to problems once `n_workers` is larger than your local cpu count as you specified it as a mini_task.

Actually, once this is a mini task, you don't really need to specify rqmts for this as it should be so small that the default will suffice (but you still can)
```suggestion
```

Otherwise consider making it a full (not-mini) task.

Reply:
I made this a full task now. The plotting can take about an hour otherwise.